artemis-server/src/main/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImpl.java

```diff
@@ -2064,6 +2072,53 @@ public Configuration setNetworkCheckPing6Command(String command) {
       return this;
    }
 
+   @Override
+   public boolean isAnalyzeCritical() {
+      return analyzeCritical;
+   }
+
+   @Override
+   public Configuration setAnalyzeCritical(boolean analyzeCritical) {
+      this.analyzeCritical = analyzeCritical;
+      return this;
+   }
+
+   @Override
+   public long getAnalyzeCriticalTimeout() {
+      return analyzeCriticalTimeout;
+   }
+
+   @Override
+   public Configuration setAnalyzeCriticalTimeout(long timeout) {
+      this.analyzeCriticalTimeout = timeout;
+      return this;
+   }
+
+   @Override
+   public long getAnalyzeCriticalCheckPeriod() {
+      if (analyzeCriticalCheckPeriod == ActiveMQDefaultConfiguration.getAnalyzeCriticalCheckPeriod()) {
```

Comment:
I was more getting at should the defaulting logic of using the timeout if the value is not set live in the config parser?